src/main/java/nl/ordina/jobcrawler/controller/YachtVacancyScraper.java

```diff
@@ -0,0 +1,150 @@
+package nl.ordina.jobcrawler.controller;
+
+import nl.ordina.jobcrawler.model.Vacancy;
+import nl.ordina.jobcrawler.service.ConnectionDocumentService;
+import nl.ordina.jobcrawler.service.Utils;
+import org.jsoup.nodes.Document;
+import org.jsoup.nodes.Element;
+import org.jsoup.select.Elements;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import java.io.IOException;
+import java.util.*;
+
+/*
+YachtVacancyScraper.java takes care of all java related vacancies on yacht.nl
+Most of the code in this class is based on the available Arabot code. Some changes were needed due to (probably) changes on the Yacht website.
+ */
+
+@Component
+public class YachtVacancyScraper implements VacancyScraper {
+    private static final String SEARCH_URL = "https://www.yacht.nl/vacatures?zoekterm=java&soortdienstverband=Detachering";
+    
+    private List<Vacancy> vacancies = new ArrayList<>();
+    
+    private final ConnectionDocumentService connectionDocumentService;
+    
+    @Autowired
+    public YachtVacancyScraper(ConnectionDocumentService connectionDocumentService) {
+        this.connectionDocumentService = connectionDocumentService;
+    }
+
+    @Override
+    public List<Vacancy> getVacancies() throws IOException {
+        /*
+        getVacancies connect to the SEARCH_URL and requests this html page.
+        If the response is not null it selects the a tags and requests the urls to the vacancy pages. The 'hours' information is only available on the main page and not on the detailed vacancy pages. Therefore I created as a (temporary) solution as hashmap where I store this data in a key->value storage (url->hours).
+         */
+        HashMap<String, String> vacancyURLs = getVacancyURLs();
+        Iterator vacancyUrl = vacancyURLs.entrySet().iterator();
+        while(vacancyUrl.hasNext()){
+            Map.Entry pair = (Map.Entry)vacancyUrl.next();
+            Document doc = connectionDocumentService.getConnection(pair.getKey().toString());
+            if(doc != null) {
+                    Vacancy vacancy = new Vacancy();
+                    vacancy.setVacancyURL(pair.getKey().toString());
+                    vacancy.setBroker("Yacht");
+                    vacancy.setHours(pair.getValue().toString());
+                    setVacancyTitle(doc, vacancy);
+                    setVacancySpecifics(doc, vacancy);
+                    setVacancyAbout(doc, vacancy);
+                    setVacancySkillSet(doc, vacancy);
+                    this.vacancies.add(vacancy);
+            }
+        }
+        return vacancies;
+    }
+
+    private HashMap<String, String> getVacancyURLs() throws IOException {
+        //  Returns a hashmap with urls and hours a week
+        HashMap<String, String> vacancyURLs = new HashMap<>();
+        int totalNumberOfPages = 1;
```

Comment:
I think this should be found somewhere, but is hardcoded here for the time being. Maybe we should create a task for this on JIRA?

Reply:
I added a new 'VacancyURLs' class which can be used in the getVacancyURLs() method. This way this method, regardless of what website will be scraped, can return a List of VacancyURLs.

In Arabot the totalNumberOfpages was set to 1 to start the loop. This variable gets updated during the first execution. I added a method to retrieve the amount of pages, but that needs some work.

Reply:
Is added to JIRA